Source/Models/HTTPResponse.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+public struct HTTPResponse {
+  public let data: NSData?
+  public let code: Int
+
+  public init(data: NSData?, response: NSURLResponse?) {
+    self.data = data
+    self.code = (response as? NSHTTPURLResponse)?.statusCode ?? 500
```

Comment:
Unfortunately, no. `NSURLResponse` and `NSHTTPURLResponse` are two different classes, and only the `HTTP` version has a `statusCode` property. Pretty dumb.